demo-extended/src/main/AndroidManifest.xml

```diff
@@ -16,6 +16,10 @@
       </intent-filter>
     </activity>
 
+    <receiver
```

Comment:
shouldn't you declare this an internal receiver to reduce visibility (not allowing other people to trigger it?)